nuxeo-drive-elasticsearch/src/main/java/org/nuxeo/drive/elasticsearch/ESAuditChangeFinder.java

```diff
@@ -227,25 +230,8 @@ protected BoolQueryBuilder getEventsClause(String category, String[] eventIds, b
 
     @Override
     public long getUpperBound() {
-        SearchRequest request = new SearchRequest(getESIndexName()).types(ElasticSearchConstants.ENTRY_TYPE)
-                                                                   .searchType(SearchType.DFS_QUERY_THEN_FETCH);
-        // TODO refactor this to use max clause
-        request.source(
-                new SearchSourceBuilder().query(QueryBuilders.matchAllQuery()).sort("id", SortOrder.DESC).size(1));
-        logSearchRequest(request);
-        SearchResponse searchResponse = getClient().search(request);
-        logSearchResponse(searchResponse);
-        List<LogEntry> entries = new ArrayList<>();
-        SearchHits hits = searchResponse.getHits();
-        ObjectMapper mapper = new ObjectMapper();
-        for (SearchHit hit : hits) {
-            try {
-                entries.add(mapper.readValue(hit.getSourceAsString(), LogEntryImpl.class));
-            } catch (IOException e) {
-                log.error("Error while reading Audit Entry from ES", e);
-            }
-        }
-        return entries.size() > 0 ? entries.get(0).getId() : -1;
+        SQLRepositoryService repositoryService = Framework.getService(SQLRepositoryService.class);
+        return getUpperBound(new HashSet<String>(repositoryService.getRepositoryNames()));
```

Comment:
Here you'll get only the VCS repository names. You should use `org.nuxeo.ecm.core.api.repository.RepositoryManager#getRepositoryNames` instead.

Reply:
:+1: